### PR TITLE
Build/test/push the concepts API in CI

### PR DESCRIPTION
**Pull request by jamieparkinson**
lemon squeezy

Only interesting bit is the change to `run_job.py` to avoid re-building scala projects when this mysterious javascript project changes.

### CODE REVIEW

**Comment by alexwlchan** [comment]
> Only interesting bit is the change to run_job.py to avoid re-building scala projects when this mysterious javascript project changes.

Q: Should we do away with that logic entirely? The entire build runs in <5m, is it worth trading a simpler build system for more EC2 minutes?

**Comment by jamieparkinson** [comment]
Not at the moment I don't think - let's just wait until we get rid of scala in this repo 😉

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -44,6 +44,27 @@ steps:
...
       queue: "scala"
 
+  - label: "Test concepts API"
+    key: "concepts-test"
+    plugins:
+      - docker-compose#v3.9.0:
+          run: concepts
+          command: ["yarn", "test"]
+
+  - label: "Publish concepts API (ECR)"
```
Comment (alexwlchan) [inline .buildkite/pipeline.yml:54]: I realise this is probably copied from the dotorg repo; could we use consistent labelling with the other tasks in this repo please?

**Comment by alexwlchan** [comment]
Okay, let's leave it for now, but I think we should consider it across the board for our Scala apps. (Possibly coupled with a breaking up of the pipeline repo.)

File: terraform/shared/ecr.tf [hunk 2]
```diff
@@ -33,6 +33,14 @@ resource "aws_ecr_repository" "api" {
   }
 }
 
+resource "aws_ecr_repository" "concepts" {
+  name = "uk.ac.wellcome/concepts"
```
Comment (alexwlchan) [inline terraform/shared/ecr.tf:37]: weco/concepts?

Or does that wreak havoc with weco-deploy?

Comment (alexwlchan) [reply terraform/shared/ecr.tf:37]: Also, is this an opportunity to start defining some better defaults around ECR repository policies for expiring old images?

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (jamieparkinson) [reply .buildkite/pipeline.yml:54]: It wasn't - that's not a straightforward thing to do as (1) the other tasks handle the test/push logic inside `run_job.py` and (2) they're very inconsistently named as it is. I've renamed some things a little but there is no way of doing this consistently with the other APIs as the steps aren't exactly analogous.

File: terraform/shared/ecr.tf (hunk 2 above)
Comment (jamieparkinson) [reply terraform/shared/ecr.tf:37]: Yeah, this naming is to satisfy weco-deploy unfortunately.

I was thinking about repo policies too - yes, we should do that, but not as part of this PR. Standardising the policies won't be harder if we have an extra repo, and I don't want this effort to get a service up and running quickly to accrue wider improvements as we do it. Rather, we should be tracking the things that it makes us notice and then fixing them as we see fit: https://github.com/wellcomecollection/platform/issues/5546

Comment (alexwlchan) [reply terraform/shared/ecr.tf:37]: Sounds good on both counts. 👍

**Review by alexwlchan — APPROVED** [review]

**Pull request merged**